pkg/controller/secret/secret_controller.go

```diff
@@ -0,0 +1,428 @@
+package secret
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	alertmanager "github.com/openshift/configure-alertmanager-operator/pkg/types"
+
+	yaml "gopkg.in/yaml.v2"
+)
+
+var log = logf.Log.WithName("controller_secret")
+
+// Add creates a new Secret Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileSecret{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("secret-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource (type "Secret").
+	// For each Add/Update/Delete event, the reconcile loop will be sent a reconcile Request.
+	err = c.Watch(&source.Kind{Type: &corev1.Secret{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcileSecret{}
+
+// ReconcileSecret reconciles a Secret object
+type ReconcileSecret struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a Secret object and makes changes based on the state read.
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileSecret) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling Secret")
+
+	// This operator is only interested in the 3 secrets listed below. Skip reconciling for all other secrets.
+	if request.Name != "alertmanager-main" && request.Name != "dms-secret" && request.Name != "pd-secret" {
+		reqLogger.Info("Skip reconcile: No changes detected to alertmanager secrets.")
+		return reconcile.Result{}, nil
+	}
+	fmt.Println("DEBUG: Started reconcile loop")
```

Comment:
Why are you using fmt.Println instead of the logger? reqLogger

Reply:
Because I found it to be a lot easier to read when I'm debugging things. Otherwise I have to scroll past a lot of output and it's harder on human eyes to read through the json formatting. Here's some example output. It helps separate out times when the operator is processing things that don't matter, vs things that a human would care about:

https://gist.githubusercontent.com/dak1n1/24a90c2ce8cfcd5f90021d05c8d70dca/raw/6027eb5ceab698557ac00e39e3aad53fe39878ef/gistfile1.txt

Reply:
I'll go ahead and make everything use log instead. Maybe I can configure the timestamps to make more sense, and clean up the amount of stuff logger is printing to make it easier to read.

Reply:
I opened an RFE to make the timestamps easier to read. I'll check it out after this PR merges. https://github.com/openshift/configure-alertmanager-operator/issues/4